### PR TITLE
Announce stream multiple channels

### LINKED ISSUE #31: Stream Announcer Module - Multiple Channels
Currently the user can only select one channel to announce when a user is streaming. We need to expand this to allow any amount.

### PR DESCRIPTION
**Pull request by consolelogreece**
Closes #31

### CODE REVIEW

File: src/Volvox.Helios.Core/Modules/StreamAnnouncer/StreamAnnouncerModule.cs [hunk 1]
```diff
@@ -43,10 +45,10 @@ public override Task Init(DiscordSocketClient client)
             // Subscribe to the GuildMemberUpdated event.
             client.GuildMemberUpdated += async (user, guildUser) =>
             {
-                var settings = await _settingsService.GetSettingsByGuild(guildUser.Guild.Id);
+                var settings = await _settingsService.GetSettingsByGuild(guildUser.Guild.Id, x => x.ChannelSettings);
 
                 if (settings != null && settings.Enabled)
-                    await CheckUser(guildUser);
+                    await CheckUser(guildUser, settings.ChannelSettings);
```
Comment (cjbose) [inline src/Volvox.Helios.Core/Modules/StreamAnnouncer/StreamAnnouncerModule.cs]: Should we have a null check for settings.ChannelSettings here?

File: src/Volvox.Helios.Core/Modules/StreamAnnouncer/StreamAnnouncerModule.cs [hunk 2]
```diff
@@ -69,94 +72,116 @@ private async Task CheckUser(SocketGuildUser user)
...
-                await AnnounceUser(user, message);
+                await AnnounceUserHandler(user, channels);
             }
 
             // User is not streaming.
             else if (user.Game == null || user.Game.Value.StreamType != StreamType.Twitch)
             {
                 // Get user from streaming list.
-                var userDataFromList = StreamingList[user.Guild.Id].FirstOrDefault(x => x.UserId == user.Id);
+                var userDataFromList = StreamingList[user.Guild.Id].Where(x => x.UserId == user.Id).ToList();
```
Comment (cjbose) [inline src/Volvox.Helios.Core/Modules/StreamAnnouncer/StreamAnnouncerModule.cs]: Now that this isn't being set to default if no matches are found, I think we need a null check here as well.

File: src/Volvox.Helios.Web/Controllers/SettingsController.cs [hunk 3]
```diff
@@ -56,42 +62,93 @@ public SettingsController(IModuleSettingsService<StreamAnnouncerSettings> stream
...
+                    ChannelId = viewModel.ChannelId
+                };
+            }
+
+            settings.RemoveMessage = viewModel.ChannelSettings.RemoveMessages;
+
+            // Save specific channel settings to the database.
+            if (viewModel.ChannelSettings.Enabled)
+            {
+                if (!isSettingsInDb)
```
Comment (cjbose) [inline src/Volvox.Helios.Web/Controllers/SettingsController.cs]: Curly braces aren't required in if statements like this, but it's my personal opinion that it looks cleaner to have them.

File: src/Volvox.Helios.Web/Controllers/SettingsController.cs [hunk 4]
```diff
@@ -56,42 +62,93 @@ public SettingsController(IModuleSettingsService<StreamAnnouncerSettings> stream
...
+                ChannelSettings = new StreamAnnouncerChannelSettingsViewModel(),
+                GuildId = guildId.ToString()
             };
 
-            var settings = await _streamAnnouncerSettingsService.GetSettingsByGuild(guildId);
+            // Get general module settings for guild, from database.
+            var settings = await _streamAnnouncerSettingsService.GetSettingsByGuild(guildId, x => x.ChannelSettings);
 
-            if (settings == null) return View(viewModel);
+            if (settings == null)
```
Comment (cjbose) [inline src/Volvox.Helios.Web/Controllers/SettingsController.cs]: Make this if statement a single line.

File: src/Volvox.Helios.Web/Controllers/SettingsController.cs [hunk 5]
```diff
@@ -56,42 +62,93 @@ public SettingsController(IModuleSettingsService<StreamAnnouncerSettings> stream
...
 
-            viewModel.ChannelId = settings.AnnouncementChannelId;
             viewModel.Enabled = settings.Enabled;
-            viewModel.RemoveMessages = settings.RemoveMessages;
+
+            // Gets first text channel's settings to prepopulate view with.
+            var defaultChannel = settings.ChannelSettings.FirstOrDefault(x => x.ChannelId == textChannels[0].Id);
+
+            // No channels setting saved, return viewmodel as is.
+            if (defaultChannel == null)
```
Comment (cjbose) [inline src/Volvox.Helios.Web/Controllers/SettingsController.cs]: Make this if statement a single line.

File: src/Volvox.Helios.Web/Controllers/SettingsController.cs [hunk 6]
```diff
@@ -114,7 +171,8 @@ public async Task<IActionResult> StreamAnnouncerSettings(ulong guildId, StreamAn
 
             var settings = await _streamerRoleSettingsService.GetSettingsByGuild(guildId);
 
-            if (settings == null) return View(viewModel);
+            if (settings == null)
```
Comment (cjbose) [inline src/Volvox.Helios.Web/Controllers/SettingsController.cs]: Make this if statement a single line.

File: src/Volvox.Helios.Web/Views/Settings/StreamAnnouncerSettings.cshtml [hunk 7]
```diff
@@ -4,27 +4,38 @@
 }
 
 <h2>Stream Announcer Settings</h2>
-<hr />
```
Comment (BillChirico) [inline src/Volvox.Helios.Web/Views/Settings/StreamAnnouncerSettings.cshtml]: Did you mean to delete this? If so we'll have to remove from the other module settings.

File: src/Volvox.Helios.Web/Views/Settings/StreamAnnouncerSettings.cshtml [hunk 8]
```diff
@@ -4,27 +4,38 @@
...
 
 <div class="row">
-    <div class="col-md-4">
+    <div class="col-md-12">
         <form method="post">
             <div asp-validation-summary="All" class="text-danger"></div>
+
+            <div style="display:flex;" class="form-group">
+                <label asp-for="Enabled" style="margin: 0 20px 0 0;" class="font-size-small"></label>
```
Comment (BillChirico) [inline src/Volvox.Helios.Web/Views/Settings/StreamAnnouncerSettings.cshtml]: Can we move this inline style to a CSS class in site.css?

File: src/Volvox.Helios.Web/Views/Settings/StreamerRoleSettings.cshtml [hunk 9]
```diff
@@ -10,16 +10,21 @@
     <div class="col-md-4">
         <form method="post">
             <div asp-validation-summary="All" class="text-danger"></div>
+
+            <div style="display:flex;" class="form-group">
+                <label asp-for="Enabled" style="margin: 0 20px 0 0;" class="font-size-small"></label>
```
Comment (BillChirico) [inline src/Volvox.Helios.Web/Views/Settings/StreamerRoleSettings.cshtml]: Remove inline style

**Pull request merged**